anomalib/data/utils/split.py

```diff
@@ -11,76 +11,128 @@
 # Copyright (C) 2022 Intel Corporation
 # SPDX-License-Identifier: Apache-2.0
 
-import random
-from typing import Optional
+from __future__ import annotations
 
-from pandas.core.frame import DataFrame
+import math
+import warnings
+from enum import Enum
+from typing import TYPE_CHECKING, List, Optional, Sequence, Tuple, Union
 
+import torch
 
-def split_normal_images_in_train_set(
-    samples: DataFrame, split_ratio: float = 0.1, seed: Optional[int] = None, normal_label: str = "good"
-) -> DataFrame:
-    """Split normal images in train set.
+if TYPE_CHECKING:
+    from anomalib.data import AnomalibDataset
 
-        This function splits the normal images in training set and assigns the
-        values to the test set. This is particularly useful especially when the
-        test set does not contain any normal images.
 
-        This is important because when the test set doesn't have any normal images,
-        AUC computation fails due to having single class.
+class Split(str, Enum):
+    """Split of a subset."""
 
-    Args:
-        samples (DataFrame): Dataframe containing dataset info such as filenames, splits etc.
-        split_ratio (float, optional): Train-Test normal image split ratio. Defaults to 0.1.
-        seed (int, optional): Random seed to ensure reproducibility. Defaults to 0.
-        normal_label (str): Name of the normal label. For MVTec AD, for instance, this is normal_label.
+    TRAIN = "train"
+    VAL = "val"
+    TEST = "test"
 
-    Returns:
-        DataFrame: Output dataframe where the part of the training set is assigned to test set.
-    """
 
-    if seed is not None:
-        random.seed(seed)
+class TestSplitMode(str, Enum):
+    """Splitting mode used to obtain subset."""
 
-    normal_train_image_indices = samples.index[(samples.split == "train") & (samples.label == normal_label)].to_list()
-    num_normal_train_images = len(normal_train_image_indices)
-    num_normal_valid_images = int(num_normal_train_images * split_ratio)
+    NONE = "none"
+    FROM_DIR = "from_dir"
+    SYNTHETIC = "synthetic"
 
-    indices_to_split_from_train_set = random.sample(population=normal_train_image_indices, k=num_normal_valid_images)
-    samples.loc[indices_to_split_from_train_set, "split"] = "test"
 
-    return samples
+class ValSplitMode(str, Enum):
```

Comment:
it would be a bit verbose, but how about `ValidationSplitMode`? Test split has the full name, maybe we could follow the same convention?

Reply:
Done. Should I also change the parameter name in the config and the variable names accordingly?

Reply:
Hmm, I'm not sure actually. `validation_split_mode` and `validation_split_ratio` are pretty verbose, which might be an annoyance for users who want to change these parameters through CLI. Also I feel that for terms starting with `val_split` it is immediately clear that 'val' stands for 'validation'.

What do you think @ashwinvaidya17 ?

Reply:
Let's go with `val_split_mode` then. I've reverted the changes to the enum, because I feel the enum should have the same name as the variable.